CIBA with redirect documentation (#84)
vippsas/vipps-login-api: vipps-login-api.md

* `error_code=old_app`: The user's Vipps app is outdated and does not support this login flow.
* `error_code=invalid_user`: No account exists, the user's account is not active or the user is in some way not eligible to use this login flow currently e.g. U15 users.

> ### CIBA - complete logged in into the merchant's web page

Reviewer: ```suggestion
### CIBA - complete login in the Vipps app with redirect to the merchant's web page
```

        "formatted": "jghj khhjhhkjh\n0603\nOSLO\nNO",
        "postal_code": "0603",
        "region": "OSLO",
        "street_address": "jghj khhjhhkjh"
      },
      "family_name": "Heyerdahl",
      "given_name": "Tor Fos",
      "name": "Tor Fos Heyerdahl",
      "other_addresses": [],
      "sid": "qwieuhwqiuhdiuwqh",
      "sub": "f350ef33-22e2-47d0-9f47-12345667"
    }
    ```

#### Authentication Request With Redirect (https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#auth_request)

The Backchannel Authentication Endpoint is listed as `backchannel_authentication_endpoint` in the configuration https://api.vipps.no/access-management-1.0/access/.well-known/openid-configuration.

##### Authentication
The following authentication methods are currently supported:
* client_secret_basic
* client_secret_post

The default token endpoint authentication method is `client_secret_basic`. It is possible to change the authentication method to `client_secret_post` in the Vipps portal.

Required parameters: `login_hint`, `scope`, `redirect_uri`, `nonce`, `code_challenge`, `requested_flow`

##### The `requested_flow` parameter (required)

> Must equal to be `login_to_webpage`.

Reviewer: ```suggestion
Must be `login_to_webpage`.
```

Note: "the binding_message value SHOULD be relatively short and use a limited set of plain text characters"

Example: `....&binding_message=4MZ-CQ3&...`

##### The `redirect_uri` parameter (required)
Redirect URL which the user agent is redirected to after finishing a login. Must be `https` in the production environment.

Example: `...&redirect_uri=https://merchant.com/callback&...`

##### The `nonce` parameter (required)
As described in the [OIDC standard](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest). String value used to associate a Client session with an ID Token, and to mitigate replay attacks. The value is passed through unmodified from the Authentication Request to the ID Token. Sufficient entropy MUST be present in the nonce values used to prevent attackers from guessing values.

##### The `code_challenge_method` parameter
Recommended value is `S256`, default value is `plain`.
This is connected with the [code_challenge parameter](#the-code_challenge-parameter-required)

##### The `code_challenge` parameter (required)
The `code_challenge` and `code_challenge_method` parameters enables support for [PKCE](https://datatracker.ietf.org/doc/html/rfc7636).

> Creating a valid value for this parameter can be a bit tricky and we recommend that you use an library like [Nimbus for Java](https://connect2id.com/products/nimbus-oauth-openid-connect-sdk)

Reviewer: ```suggestion
Creating a valid value for the `code_challenge` can be a bit tricky and we recommend that you use a library like [Nimbus for Java](https://connect2id.com/products/nimbus-oauth-openid-connect-sdk)
```
